iptables/iptables.go

```diff
@@ -72,6 +72,20 @@ type IPTables struct {
 	mode           string // the underlying iptables operating mode, e.g. nf_tables
 }
 
+// Stat represents a structured statistic entry.
+type Stat struct {
+	Packets     string `json:"pkts"`
+	Bytes       string `json:"bytes"`
+	Target      string `json:"target"`
+	Protocol    string `json:"prot"`
+	Opt         string `json:"opt"`
+	Input       string `json:"in"`
+	Output      string `json:"out"`
+	Source      string `json:"source"`
```

Comment:
I forget - can these be hostnames, or do they have to be IP addresses? If they have to be IP addresses, can you parse them accordingly?

Reply:
The method `Stats(...)` uses the `-n` flag. From the man pages:

```  --numeric	-n		numeric output of addresses and ports```

This shows the source and destination in CIDR format. Below is some output:

```
$ iptables --version
iptables v1.4.21
$ iptables -t filter -L INPUT -n -v -x
Chain INPUT (policy ACCEPT 12 packets, 857 bytes)
    pkts      bytes target     prot opt in     out     source               destination
       2      128 ACCEPT     tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp dpt:3332
$ iptables -t filter -L INPUT  -v -x
Chain INPUT (policy ACCEPT 43 packets, 3073 bytes)
    pkts      bytes target     prot opt in     out     source               destination
       2      128 ACCEPT     tcp  --  any    any     anywhere             anywhere             tcp dpt:mcs-mailsvr
```

From this, we know it will be CIDR notation IP addresses.

We can call this struct `NumericalStats` to be more accurate. Since it's already granular, I don't think this is necessary.